src/main/java/thePackmaster/events/BlackMarketDealerEvent.java

```diff
@@ -0,0 +1,174 @@
+//
+// Source code recreated from a .class file by IntelliJ IDEA
+// (powered by Fernflower decompiler)
+//
+
+package thePackmaster.events;
+
+import BattleTowers.BattleTowers;
+import BattleTowers.relics.OttosDeck;
+import com.megacrit.cardcrawl.cards.AbstractCard;
+import com.megacrit.cardcrawl.cards.CardGroup;
+import com.megacrit.cardcrawl.cards.curses.Shame;
+import com.megacrit.cardcrawl.core.CardCrawlGame;
+import com.megacrit.cardcrawl.core.Settings;
+import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
+import com.megacrit.cardcrawl.events.AbstractImageEvent;
+import com.megacrit.cardcrawl.localization.EventStrings;
+import com.megacrit.cardcrawl.vfx.cardManip.ShowCardAndObtainEffect;
+import thePackmaster.SpireAnniversary5Mod;
+
+import java.util.Iterator;
+
+import static BattleTowers.BattleTowers.makeID;
+import static thePackmaster.SpireAnniversary5Mod.makeID;
+
+public class BlackMarketDealerEvent extends AbstractImageEvent {
```

Comment:
Was this a WIP version of an event to get Pack Rip or something like that? Looks to me like it shouldn't be part of this PR -- everything else looks good, so if you remove this, I can merge the rest.

Reply:
Ah crap, I thought I was on another branch already.  Well I just pushed the full event, so this is now "two fixes and a giant event" PR.